plugin/storage/integration/cassandra_test.go

```diff
@@ -98,8 +100,10 @@ func (s *CassandraStorageIntegration) initializeDependencyReaderAndWriter(t *tes
 }
 
 func TestCassandraStorage(t *testing.T) {
+	defer testutils.VerifyGoLeaksOnce(t)
 	SkipUnlessEnv(t, "cassandra")
 	s := newCassandraStorageIntegration()
-	s.initializeCassandra(t)
+	f := s.initializeCassandra(t)
 	s.RunAll(t)
+	require.NoError(t, f.Close())
```

Comment:
this is already done in L83, why do we need to close again?

Reply:
your explanation does not make sense to me. The call to VerifyNone is the first statement in the test. Clearly it cannot run at that point, so it must be scheduling itself for the test clean-up phase (that's just a speculation without looking at the implementation).

Reply:
Got it! Will search about it and get back to you!

Reply:
> Please have a look on it, if this issue is correct: [uber-go/goleak#53](https://github.com/uber-go/goleak/issues/53)
 I don't think the kind of scheduling you are saying take place in `goleak` as I can't see any implementation of this kind in its code!

Reply:
here's what we can do: instead of using `defer` invoke this from Cleanup explicitly
```
t.Cleanup(func() { testutils.VerifyGoLeaksOnce(t) })
```
This will have similar impact but will also ensure that this runs as the very _last_ step of the clean-up (these lambdas are unwound in reverse order).